cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/ConcurrentStreamWriter.java

```diff
@@ -111,6 +111,11 @@
     this.createLock = new ReentrantLock();
   }
 
+  public void close(Id.Stream streamId) throws IOException, NotFoundException {
+    EventQueue eventQueue = getEventQueue(streamId);
```

Comment:
hum.. it's slightly more complicated than that, because there could be races from other thread trying to create it. So you'll need to acquire the `createLock` before doing `Closeables.close(eventQueues.get(streamId), false)`